include/SugarObjects/templates/basic/Basic.php

```diff
@@ -82,76 +88,100 @@ public function get_summary_text()
-        if ($confirmOptIn && $sugar_config['email_enable_confirm_opt_in']) {
-            $tickTitle = $app_strings['LBL_CONFIRM_OPT_IN_TITLE'];
-            $tickHtml = '<span class="confirm-opt-in-tick" title="' . $tickTitle . '">&#10004;</span>';
+
+    /**
+     * @global array $sugar_config
+     * @global \LoggerManager $log
+     * @param string $emailField
+     * @return string
+     */
+    public function getOptInStatusFromSugarField($emailField)
+    {
+        $emailAddress = $this->fromSugarEmailAddressField($emailField);
+
+
+        if($emailAddress !== null && !in_array($this->module_name, self::$doNotDisplayOptInTickForModule, true)) {
+            if($emailAddress->invalid_email) {
+                return 'INVALID_EMAIL';
+            }
+
+            if ($emailAddress->opt_out == "1") {
+                return 'OPT_OUT';
+            }
+
+            if ($emailAddress->confirm_opt_in == "1") {
+                return "OPT_IN_PENDING_EMAIL_CONFIRMED";
+            } elseif(!empty($emailAddress->opt_in_email_created)) {
+                return "OPT_IN_PENDING_EMAIL_SENT";
+            } elseif(empty($emailAddress->opt_in_email_created)) {
+                return "OPT_IN_PENDING_EMAIL_NOT_SENT";
+            }
         }
 
-        return $tickHtml;
+        // Otherwise
+        return 'UNKNOWN_OPT_IN_STATUS';
     }
 
     /**
      *
      * @global array $sugar_config
      * @global \LoggerManager $log
      * @param string $emailField
-     * @return boolean
+     * @return \EmailAddress|\SugarBean
      * @throws RuntimeException
      * @throws InvalidArgumentException
      */
-    public function getEmailAddressConfirmOptIn($emailField)
+    public function fromSugarEmailAddressField($emailField)
     {
-        $this->getEmailAddressValidateArguments($emailField);
+        $this->validateSugarEmailAddressField($emailField);
 
         global $sugar_config;
 
+        /** @var EmailAddress $emailAddressBean */
+        $emailAddressBean = BeanFactory::getBean('EmailAddresses');
+
         if (!$sugar_config['email_enable_confirm_opt_in']) {
             global $log;
             $log->warn('Confirm Opt In is not enabled.');
-            return true;
+            $emailAddressBean->confirm_opt_in = true;
+            return $emailAddressBean;
         }
 
-        $emailAddressId = $this->getEmailAddressId($emailField);
-        $emailAddress = BeanFactory::getBean('EmailAddresses', $emailAddressId);
-        $confirmOptIn = $emailAddress->confirm_opt_in;
-
-        return $confirmOptIn;
+        $emailAddressId = $this->getIdFromSugarEmailAddressField($emailField);
+        return $emailAddressBean->retrieve($emailAddressId);
     }
 
     /**
      *
      * @global \LoggerManager $log
      * @param string $emailField
      * @return string|null EmailAddress ID or null on error
-     * @throws RuntimeException
-     * @throws InvalidArgumentException
```

Comment:
Update php doc: This method still can throws `InvalidArgumentException`.

Reply:
updated php doc